pkg/operator/ceph/file/controller.go

```diff
@@ -270,3 +277,20 @@ func (c *FilesystemController) releaseOrchestrationLock() {
 	c.orchestrationMutex.Unlock()
 	logger.Debugf("Released lock for filesystem orchestration")
 }
+
+func updateCephFilesystemStatus(name, namespace, status string, context *clusterd.Context) {
+	updatedCephFilesystem, err := context.RookClientset.CephV1().CephFilesystems(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("Unable to update the cephObjectStore %s status %v", updatedCephFilesystem.GetName(), err)
```

Comment:
Need to return here after the err, otherwise `updatedCephFilesystem` could be nil. This also applies to the status of other CR types.

As a general pattern, if we don't return after an error is returned we need to be clear in a comment about why that is expected to continue even with the error. But usually it's a bug if we don't return.

Reply:
yup updated. I think I was sleepy while refactoring. :)